### PR TITLE
prismarine-entity 2.0

### PR DESCRIPTION
**Pull request by KaffinPX**
Adds so much things <3

### CODE REVIEW

**Comment by Lennster1** [comment]
dang that's some pretty good work nice job!

**Comment by KaffinPX** [comment]
> dang that's some pretty good work nice job!

Thanks but i think its not completed, i need some reviews.

**Comment by u9g** [comment]
This is useless, you just added my pr that was closed for a reason, very small tests that don’t even test correctly, merged some dev dep updates for no reason and you changed a bit of ci for literally no reason.

**Comment by KaffinPX** [comment]
I wrapped it with a class, look again.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,24 +2,36 @@ name: CI
...
       with:
-        node-version: ${{ matrix.node-version }}
-    - run: npm install
-    - run: npm test
\ No newline at end of file
+        node-version: 14.x
+    - name: Install Dependencies
+      run: npm install
+    - name: Start Tests
+      run: npm run test
```
Comment (Saiv46) [inline .github/workflows/ci.yml]: Please do not edit configuration files for no reason

File: README.md [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
+Represent a minecraft entity with an api.
 
 ## Usage
 
 ```js
-var Entity=require("prismarine-entity");
+let prismarineEntity = require("prismarine-entity")
 
-var entity=new Entity(0);
+let entity = new prismarineEntity(0)
```
Comment (Saiv46) [inline README.md:12]: Classes should be named CamelCase.

File: index.js [hunk 3]
```diff
@@ -1,30 +1,7 @@
...
-  this.effects = {}
-  // 0 = held item, 1-4 = armor slot
-  this.equipment = new Array(5)
-  this.heldItem = this.equipment[0] // shortcut to equipment[0]
-  this.isValid = true
-  this.metadata = []
+if (typeof process !== 'undefined' && parseInt(process.versions.node.split('.')[0]) < 14) {
+  console.error('Your node version is currently', process.versions.node)
+  console.error('Please update it to a version >= 14.x.x from https://nodejs.org/')
+  process.exit(1)
```
Comment (Saiv46) [inline index.js:4]: This should not be here, better add `engine.node` property in `package.json`

File: test/internalTest.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/* eslint-env mocha */
+
+const versions = require('../src/version')
+
+for (const supportedVersion of versions.testedVersions) {
+  describe(`prismarineEntity ${supportedVersion}`, function () {
+    const PrismarineEntity = require('../')(supportedVersion)
```
Comment (Saiv46) [inline test/internalTest.js:7]: This thing does not support parallel testing, please look at how other PrismarineJS projects are being tested.

File: test/internalTest.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    it('reads metadata', (done) => {
+      let Entity = new PrismarineEntity(0, true)
+      if (Entity.isValid === true) {
+        if (Entity.isOnFire() === false) {
+          done()
+        } else {
+          throw new Error('Entity defaultly doesnt on fire!?')
+        }
+      } else {
+        throw new Error('Entity is not valid!?')
```
Comment (Saiv46) [inline test/internalTest.js:26]: Bruh, do you use any test libraries?

**Review by Saiv46 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (KaffinPX) [reply .github/workflows/ci.yml]: I added tests.

File: README.md (hunk 2 above)
Comment (KaffinPX) [reply README.md:12]: Right

File: test/internalTest.js (hunk 5 above)
Comment (KaffinPX) [reply test/internalTest.js:26]: Yeah it uses Mocha

**Pull request closed without merging**